src/main/scala/mesosphere/marathon/core/task/jobs/impl/ExpungeOverdueLostTasksActor.scala

```diff
@@ -0,0 +1,77 @@
+package mesosphere.marathon.core.task.jobs.impl
+
+import akka.actor.{ ActorLogging, Props, Cancellable, Actor }
+import akka.pattern.pipe
+import com.google.inject.Provider
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.core.base.Clock
+import mesosphere.marathon.core.task.bus.MesosTaskStatus.TemporarilyUnreachable
+import mesosphere.marathon.core.task.jobs.TaskJobsConfig
+import mesosphere.marathon.core.task.tracker.TaskTracker
+import mesosphere.marathon.core.task.tracker.TaskTracker.AppTasks
+import mesosphere.marathon.core.task.update.TaskStatusUpdateProcessor
+import mesosphere.marathon.state.PathId
+import org.apache.mesos.Protos.TaskState
+import org.apache.mesos.{ Protos => mesos }
+import scala.concurrent.duration._
+
+class ExpungeOverdueLostTasksActor(
+    clock: Clock,
+    config: TaskJobsConfig,
+    taskTracker: TaskTracker,
+    updateProcessorProvider: Provider[TaskStatusUpdateProcessor]) extends Actor with ActorLogging {
+
+  import ExpungeOverdueLostTasksActor._
+  implicit val ec = context.dispatcher
+
+  var tickTimer: Option[Cancellable] = None
+  lazy val updateProcessor = updateProcessorProvider.get()
+
+  override def preStart(): Unit = {
+    log.info("ExpungeOverdueLostTasksActor has started")
+    tickTimer = Some(context.system.scheduler.schedule(config.taskLostExpungeInitialDelay,
+      config.taskLostExpungeInterval, self, Tick))
+  }
+
+  override def postStop(): Unit = {
+    tickTimer.foreach(_.cancel())
+    log.info("ExpungeOverdueLostTasksActor has stopped")
+  }
+
+  override def receive: Receive = {
+    case Tick                             => taskTracker.tasksByApp() pipeTo self
+    case TaskTracker.TasksByApp(appTasks) => filterLostGCTasks(appTasks).foreach(expungeLostGCTask)
+  }
+
+  def expungeLostGCTask(task: MarathonTask): Unit = {
+    log.warning(s"Task ${task.getId} is lost longer than ${config.taskLostExpungeGC} and will be expunged.")
+    val update = task.getStatus.toBuilder
+      .setState(TaskState.TASK_ERROR)
+      .setReason(mesos.TaskStatus.Reason.REASON_TASK_UNKNOWN)
+      .setTimestamp((clock.now().toDateTime.getMillis / 1000).toDouble)
+      .build()
+    updateProcessor.publish(update, ack = false)
```

Comment:
Can you add a note explaining why we publish a status update here? I'm afraid people might find this disturbing.